keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -137,34 +136,39 @@ class SqlVerificationRepository(
         .fetchOneInto<VerificationState>()
     }
 
-  override fun getStates(context: VerificationContext): Map<String, VerificationState>
-    = with(context) {
-    when {
-      verifications.isEmpty() -> emptyMap() // Optimization: don't hit the db if we know there are no entries
-      else -> jooq.select(
-        VERIFICATION_STATE.VERIFICATION_ID,
-        VERIFICATION_STATE.STATUS,
-        VERIFICATION_STATE.STARTED_AT,
-        VERIFICATION_STATE.ENDED_AT
-      )
-        .from(VERIFICATION_STATE)
-        .where(VERIFICATION_STATE.ENVIRONMENT_UID.eq(environmentUid))
-        .and(VERIFICATION_STATE.ARTIFACT_UID.eq(artifact.uid))
-        .and(VERIFICATION_STATE.ARTIFACT_VERSION.eq(version))
-        .fetch()
-        .associate { (id, status, started_at, ended_at) -> Pair(id, VerificationState(status, started_at, ended_at)) }
+  override fun getStates(context: VerificationContext): Map<String, VerificationState> =
+    with(context) {
+      when {
+        verifications.isEmpty() -> emptyMap() // Optimization: don't hit the db if we know there are no entries
+        else -> jooq.select(
+          VERIFICATION_STATE.VERIFICATION_ID,
+          VERIFICATION_STATE.STATUS,
+          VERIFICATION_STATE.STARTED_AT,
+          VERIFICATION_STATE.ENDED_AT,
+          VERIFICATION_STATE.METADATA
+        )
+          .from(VERIFICATION_STATE)
+          .where(VERIFICATION_STATE.ENVIRONMENT_UID.eq(environmentUid))
+          .and(VERIFICATION_STATE.ARTIFACT_UID.eq(artifact.uid))
+          .and(VERIFICATION_STATE.ARTIFACT_VERSION.eq(version))
+          .fetch()
+          .associate { (id, status, started_at, ended_at, metadata) ->
+            id to VerificationState(status, started_at, ended_at, metadata)
+          }
+      }
     }
-  }
 
   override fun updateState(
     context: VerificationContext,
     verification: Verification,
-    status: VerificationStatus
+    status: VerificationStatus,
+    metadata: Map<String, Any?>?
   ) {
     with(context) {
       jooq
         .insertInto(VERIFICATION_STATE)
         .set(VERIFICATION_STATE.STATUS, status)
+        .set(VERIFICATION_STATE.METADATA, metadata)
```

Comment:
I thought I saw a comment somewhere that we'd only update the metadata if the parameter is not null?
```suggestion
        .apply { if (metadata != null) set(VERIFICATION_STATE.METADATA, metadata) }
```

Reply:
`JSONConverter` implements that behavior already